### PR TITLE
reporting queries reorganized, returning better data

### PR DESCRIPTION
**Pull request by ScottAP108**
- unrelated to branch:  improved regex in `Format` for when PW wants to embed a link to an external site within text using the collection manager. /http/ was insufficient

- reorganized `resources/report` so that queries are a bit more flexible and are retrieved via function rather than object property

- reporting queries updated to return results based on all purchased shares within a time period instead of just new members created. Filtering by season returns all purchases of that season, and filtering by year returns all purchases for all seasons of the selected year.

- multiple rows allowed for each person to give better information for member info query. Share label and signup date included in rows so that it can be seen what shares were purchased and when.

- updated `handleExport` to handle all query types (was hardcoded to expect from and to query keys)

### CODE REVIEW

File: client/js/Format.js [hunk 1]
```diff
@@ -128,7 +128,7 @@ module.exports = {
 
         replacement = /email/i.test( target )
             ? `<a href="mailto:${value}" class="link">${key}</a>`
-            : /http/.test( value )
+            : /[-a-zA-Z0-9@:%_\+.~#?&//=]{2,256}\.[a-z]{2,4}\b(\/[-a-zA-Z0-9@:%_\+.~#?&//=]*)?/gi.test( value )
```
Comment (cbaron) [inline client/js/Format.js]: This is used by the dynamic content rendering system to decide what to do with a value?

Comment (ScottAP108) [reply client/js/Format.js]: Right. I believe we settled on {{text:link}} as the format. I forgot to inform PW on that before, so I will when I next email them.

File: resources/report.js [hunk 2]
```diff
@@ -3,17 +3,46 @@ var Base = require('./__proto__'),
...
+        return { values, whereString }
+    },
+
     handleExport( model ) {
+        const { values, whereString } = this.getValuesAndWhere( model ),
+            fileString = Object.keys( this.query ).filter( key => ![ 'id', 'export', 'shareId' ].includes( key ) ).map( key => this.query[key] ).join('_')
+
+        let query = model.query( whereString )
+
+        values.forEach( ( val, i ) => { query = query.replace( `$${i+1}`, `'${val}'` ) } )
```
Comment (cbaron) [inline resources/report.js]: This is pretty dangerous.  https://github.com/cbaron/patchwork/issues/59

Let's leave it for now, but make it secure before launch.  Either we need to actually use sql parameters, or validate the parameters.

Comment (cbaron) [reply resources/report.js]: Yes, I realize it was dangerous before you touched it.

**Pull request merged**